packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,258 @@
+        'yarn',
+        ['config', 'set', 'registry', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa('yarn', ['config', 'get', 'registry'], execaOptions);
+    }
+    await execa('yarn', ['cache', 'clean'], execaOptions);
+  } else if (packageManager === 'pnpm') {
+    await execa('npm', ['install', '-g', packageManager], {
+      stdio: 'inherit',
+    });
+    await execa(packageManager, ['--version']);
+    await execa(packageManager, [
+      'config',
+      'set',
+      'registry',
+      'http://localhost:4873',
+    ]);
+    await execa(packageManager, ['config', 'get', 'registry']);
+  }
+
+  return { packageManagerExecutable, packageManager };
+};
+
+void describe('installs expected packages and scaffolds expected files', async () => {
+  let branchBackendIdentifier: BackendIdentifier;
+  let testBranch: TestBranch;
+  let cfnClient: CloudFormationClient;
+  const tempDir = await fsp.mkdtemp(
+    path.join(os.tmpdir(), 'test-create-amplify')
+  );
+  const { packageManagerExecutable, packageManager } =
+    await setupPackageManager(tempDir);
+
+  before(async () => {
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
+    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+
+    cfnClient = new CloudFormationClient(e2eToolingClientConfig);
+    testBranch = await amplifyAppPool.createTestBranch();
+    branchBackendIdentifier = {
+      namespace: testBranch.appId,
+      name: testBranch.branchName,
+      type: 'branch',
+    };
+  });
+
+  after(async () => {
+    await cfnClient.send(
+      new DeleteStackCommand({
+        StackName: BackendIdentifierConversions.toStackName(
+          branchBackendIdentifier
+        ),
+      })
+    );
+    await fsp.rm(tempDir, { recursive: true });
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  void it(`starting project`, async () => {
+    await execa(
+      packageManager.startsWith('yarn') ? 'yarn' : packageManager,
+      ['create', 'amplify', '--yes'],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    const amplifyPathPrefix = path.join(tempDir, 'amplify');
+
+    const pathPrefix = path.join(tempDir, 'amplify');
+
+    const files = await glob(path.join(amplifyPathPrefix, '**', '*'), {
+      // eslint-disable-next-line spellcheck/spell-checker
+      nodir: true,
+      windowsPathsNoEscape: true,
+      ignore: ['**/node_modules/**', '**/yarn.lock'],
+    });
+
+    const expectedAmplifyFiles = [
+      path.join('auth', 'resource.ts'),
+      'backend.ts',
+      path.join('data', 'resource.ts'),
+      'package.json',
+      'tsconfig.json',
+    ];
+
+    assert.deepStrictEqual(
+      files.sort(),
+      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+    );
+
+    if (packageManager === 'yarn-modern') {
```

Comment:
This part can't be moved to `setupPackageManager`, because has to be done after `create-amplify`

Reply:
In that case I suggest to create `continuePackageManagerSetupAfterCreateAmplify()` to host this logic.

This however makes me wonder why do we need this. I.e. is create amplify flow missing some steps or do we need to do this because of our testing strategy ?

Reply:
Good point. It turned out after we use `node_modules` for `yarn-modern`, we don't need to add most of the dependencies. [65df176](https://github.com/aws-amplify/amplify-backend/pull/901/commits/65df1765a01a3e3a0e7e004022ca9765b0fe8f17)

Reply:
Removed all of them 038ac85e15